.github/workflows/eval-lib-tests.yml

```diff
@@ -0,0 +1,25 @@
+name: "Building Nixpkgs lib-tests"
+
+permissions:
+  contents: read
+
+on:
+  pull_request_target:
+    paths:
+      - 'lib/**'
+jobs:
+  nixpkgs:
+    name: nixpkgs-lib-tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          # pull_request_target checks out the base branch by default
+          ref: refs/pull/${{ github.event.pull_request.number }}/merge
```

Comment:
We should standardise on using `get-merge-commit.sh`, check out the [docs](https://github.com/NixOS/nixpkgs/tree/master/ci#get-merge-commitsh-github_repo-pr_number). It's much more reliable than `/merge`.

```suggestion
      # Important: Because of `pull_request_target`, this doesn't check out the PR,
      # but rather the base branch of the PR, which is needed so we don't run untrusted code
      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
        with:
          path: base
          sparse-checkout: ci
      - name: Resolving the merge commit
        env:
          GH_TOKEN: ${{ github.token }}
        run: |
          if mergedSha=$(base/ci/get-merge-commit.sh ${{ github.repository }} ${{ github.event.number }}); then
            echo "Checking the merge commit $mergedSha"
            echo "mergedSha=$mergedSha" >> "$GITHUB_ENV"
          else
            # Skipping so that no notifications are sent
            echo "Skipping the rest..."
          fi
          rm -rf base
      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
        # Add this to _all_ subsequent steps to skip them
        if: env.mergedSha
        with:
          ref: ${{ env.mergedSha }}
```

At some point we should turn this into a [reusable workflow](https://docs.github.com/en/actions/sharing-automations/reusing-workflows)

Reply:
Let's make this a reusable workflow (DRY)